src/modules/create_institution/app/create_institution_controller.ts

```diff
@@ -0,0 +1,58 @@
+import { CreateInstitutionUsecase } from "./create_institution_usecase";
+import { Institution } from "../../../core/structure/entities/Institution";
+
+import { EntityError } from '../../../core/helpers/errors/EntityError';
+import { BadRequest, ParameterError, InternalServerError } from '../../../core/helpers/http/http_codes';
+import { Created, HttpRequest, HttpResponse } from '../../../core/helpers/http/http_codes';
+import { MissingParameter } from '../../../core/helpers/errors/ModuleError';
+
+export class CreateInstitutionController {
+    usecase: CreateInstitutionUsecase;
+
+    constructor(usecase: CreateInstitutionUsecase) {
+        this.usecase = usecase;
+    }
+
+    async execute(request: HttpRequest): Promise<HttpResponse> {
+        try {
+            if (!request) {
+                throw new MissingParameter("Request");
+            }
+            if (!request.body || !request.body.body) { 
+                throw new MissingParameter("Body");
+            }
+            if (!request.headers) {
+                throw new MissingParameter("Headers");
+            }
+
+            const institutionData = request.body.body;
+
+            if (!institutionData.id || !institutionData.name || !institutionData.description || !institutionData.email || !institutionData.country || !institutionData.images || !institutionData.social_medias) {
+                throw new MissingParameter("InstitutionData");
+            }
+
+            const institution = new Institution({
+                id: institutionData.id,
+                name: institutionData.name,
+                description: institutionData.description,
+                email: institutionData.email,
+                country: institutionData.country,
+                images: institutionData.images,
+                social_medias: institutionData.social_medias
+            });
```

Comment:
This logic of creating a entity Institution, need to be implemented in the usecase, the controller, u only check if the request.body is not null, and them pass to usecase

Reply:
so the usecase should be responsible for creating the institution?

Reply:
Yes, the use case is responsible for all the logic and rules! The controller is used only to verify if the needed request is valid and them, he passes the body and headers, for exemple, to the use case!